alerts/kube_apiserver.libsonnet

```diff
@@ -88,7 +88,7 @@ local utils = import '../lib/utils.libsonnet';
           {
             alert: 'KubeAggregatedAPIErrors',
             expr: |||
-              sum by(name, namespace, %(clusterLabel)s)(increase(aggregator_unavailable_apiservice_total{%(kubeApiserverSelector)s}[10m])) > 4
+              sum by (instance) (max by(name, namespace, %(clusterLabel)s)(increase(aggregator_unavailable_apiservice_total{%(kubeApiserverSelector)s}[10m]))) > 4
```

Comment:
According to the [docs](https://kubernetes.io/docs/reference/instrumentation/metrics/) for this metric, and from what I can see in an example dataset, the labels available are:

- instance
- job
- cluster
- name (e.g. `v1beta1.external.metrics.k8s.io`)
- reason (e.g. `FailedDiscoveryCheck`)

I see no sign of `namespace`, so that label should be removed. `reason` seems useful so that should be added.

I've modified your query to remove the `max`, as it is not needed after all (I had originally mistaken the meaning of the word `Aggregated` in this case: it's referring to the [Kubernetes API Aggregate Layer](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/) not "aggregation in PromQL"). So I moved the `instance` label to be part of the `sum`.

I don't understand why `increase(...[10m])` is used, as all that does is report the same error for longer. I think this should be changed to `increase(...[1m])` and combined with the `for: 10m` as per your suggestion, that should be more robust handling.

Equally, the `> 4` seems kinda random/magic, with the correct 10-minute handling, surely you want to know about **any** error, not just the fact that you had more than 4.

```suggestion
              sum by (%(clusterLabel)s, instance, name, reason)(increase(aggregator_unavailable_apiservice_total{%(kubeApiserverSelector)s}[1m])) > 0
```